### ISSUE TITLE
--dns is ignored (at least for wireguard connections)

### ISSUE THREAD

**Issue by dani-jozsef** [jamesmcm/vopono#68]
Another bug I found while trying to run dnscrypt-proxy inside a namespace and use it for resolution... Here's an example debug output leading up to the wrong DNS server being set:
```
$ vopono -v exec --dns 8.8.8.8 --provider mullvad --server sweden "google-chrome"
 2021-03-09T15:27:28.194Z DEBUG vopono::pulseaudio > Setting PULSE_SERVER to /run/user/1000/pulse/native
 2021-03-09T15:27:28.194Z INFO  vopono::util       > Calling sudo for elevated privileges, current user will be used as default user
 2021-03-09T15:27:28.194Z DEBUG vopono::util       > Args: ["vopono", "-v", "exec", "--dns", "8.8.8.8", "--provider", "mullvad", "--server", "sweden", "google-chrome"]
[sudo] password for dani: 
 2021-03-09T15:27:34.338Z DEBUG vopono::pulseaudio > Setting PULSE_SERVER to /run/user/1000/pulse/native
 2021-03-09T15:27:34.341Z DEBUG vopono::util       > Existing namespaces: []
 2021-03-09T15:27:34.342Z DEBUG vopono::network_interface > ip addr
 2021-03-09T15:27:34.344Z DEBUG vopono::exec              > Interface: wlp61s0
 2021-03-09T15:27:34.347Z INFO  vopono::util              > Chosen config: /home/dani/.config/vopono/mv/wireguard/sweden-se3.conf
 2021-03-09T15:27:34.349Z DEBUG vopono::util              > Existing namespaces: []
 2021-03-09T15:27:34.349Z DEBUG vopono::util              > ip netns add vopono_mv_sweden
 2021-03-09T15:27:34.352Z INFO  vopono::netns             > Created new network namespace: vopono_mv_sweden
 2021-03-09T15:27:34.356Z DEBUG vopono::util              > Existing interfaces: 8: vethfd4843d@if7: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br-12da57f1c6db state UP group default 
    link/ether ee:4e:2b:1f:ed:60 brd ff:ff:ff:ff:ff:ff link-netnsid 0
    inet6 fe80::ec4e:2bff:fe1f:ed60/64 scope link 
       valid_lft forever preferred_lft forever

 2021-03-09T15:27:34.358Z DEBUG vopono::util              > Assigned IPs: []
 2021-03-09T15:27:34.358Z DEBUG vopono::netns             > ip netns exec vopono_mv_sweden ip addr add 127.0.0.1/8 dev lo
 2021-03-09T15:27:34.363Z DEBUG vopono::netns             > ip netns exec vopono_mv_sweden ip link set lo up
 2021-03-09T15:27:34.367Z DEBUG vopono::veth_pair         > NetworkManager detected, adding mv_sweden_d to unmanaged devices
 2021-03-09T15:27:34.368Z DEBUG vopono::util              > nmcli connection reload
 2021-03-09T15:27:34.392Z DEBUG vopono::util              > ip link add mv_sweden_d type veth peer name mv_sweden_s
 2021-03-09T15:27:34.393Z DEBUG vopono::util              > ip link set mv_sweden_d up
 2021-03-09T15:27:34.395Z DEBUG vopono::util              > ip link set mv_sweden_s netns vopono_mv_sweden up
 2021-03-09T15:27:34.409Z DEBUG vopono::util              > ip addr add 10.200.1.1/24 dev mv_sweden_d
 2021-03-09T15:27:34.411Z DEBUG vopono::netns             > ip netns exec vopono_mv_sweden ip addr add 10.200.1.2/24 dev mv_sweden_s
 2021-03-09T15:27:34.413Z DEBUG vopono::netns             > ip netns exec vopono_mv_sweden ip route add default via 10.200.1.1 dev mv_sweden_s
 2021-03-09T15:27:34.415Z INFO  vopono::netns             > IP address of namespace as seen from host: 10.200.1.2
 2021-03-09T15:27:34.415Z INFO  vopono::netns             > IP address of host as seen from namespace: 10.200.1.1
 2021-03-09T15:27:34.415Z DEBUG vopono::util              > iptables -t nat -A POSTROUTING -s 10.200.1.0/24 -o wlp61s0 -j MASQUERADE
 2021-03-09T15:27:34.425Z DEBUG vopono::util              > iptables -I FORWARD -i mv_sweden_d -o wlp61s0 -j ACCEPT
 2021-03-09T15:27:34.441Z DEBUG vopono::util              > iptables -I FORWARD -o mv_sweden_d -i wlp61s0 -j ACCEPT
 2021-03-09T15:27:34.453Z DEBUG vopono::util              > sysctl -q net.ipv4.ip_forward=1
 2021-03-09T15:27:34.458Z DEBUG vopono::wireguard         > Deserializing: 193.138.218.74 to Vec<IpAddr>
 2021-03-09T15:27:34.458Z DEBUG vopono::wireguard         > TOML config: WireguardConfig { interface: WireguardInterface { private_key: "mOAXMKxkcBDEFdY68I9iUUCgpg5OtJGUZzyibI0Ok14=", address: [10.71.107.104/32, fc00:bbbb:bbbb:bb01::8:6b67/128], dns: [193.138.218.74] }, peer: WireguardPeer { public_key: "5JMPeO7gXIbR5CnUa/NPNK4L5GqUnreF0/Bozai4pl4=", allowed_ips: [0.0.0.0/0, ::/0], endpoint: 185.213.154.66:51820 } }
 2021-03-09T15:27:34.458Z DEBUG vopono::netns             > ip netns exec vopono_mv_sweden ip link add mv_sweden type wireguard
 2021-03-09T15:27:34.460Z DEBUG vopono::netns             > ip netns exec vopono_mv_sweden wg setconf mv_sweden /tmp/vopono_nft.conf
 2021-03-09T15:27:34.471Z DEBUG vopono::netns             > ip netns exec vopono_mv_sweden ip -4 address add 10.71.107.104/32 dev mv_sweden
 2021-03-09T15:27:34.473Z DEBUG vopono::netns             > ip netns exec vopono_mv_sweden ip -6 address add fc00:bbbb:bbbb:bb01::8:6b67/128 dev mv_sweden
 2021-03-09T15:27:34.475Z DEBUG vopono::netns             > ip netns exec vopono_mv_sweden ip link set mtu 1420 up dev mv_sweden
 2021-03-09T15:27:34.488Z DEBUG vopono::dns_config        > Setting namespace vopono_mv_sweden DNS server to 193.138.218.74
```

**Comment by jamesmcm** [comment]
Thanks, it seems like the provider specific DNS is taking precedence in that case, as it is present in the wg-quick config file. I'll take a look.

**Comment by jamesmcm** [comment]
Should be fixed in version [0.6.8](https://github.com/jamesmcm/vopono/releases/tag/0.6.8) - thanks for reporting it!
